### PR TITLE
Support multiple jet tables, full jets, and jet matching

### PR DESCRIPTION
**Pull request by raymondEhlers**
A variety of developments for the JE framework including:

- Support for multiple jet tables with constituents
- Support for multiple jet finders, writing to multiple jet tables
- Support for full jets
- Implementation of jet matching algorithm
- Writing results of jet matching to jet tables

Will also cleanup the commit history when ready to commit

@nzardosh

### CODE REVIEW

**Comment by raymondEhlers** [comment]
I've resolved all of the issues discussed at today's meeting, so I'm changing it from draft to standard PR.

The changes of most interest are probably:

- The macros in `Analysis/DataModel/include/AnalysisDataModel/Jet.h`
- The `WorkflowSpec` definition in `Analysis/Tasks/PWGJE/jetfinder.cxx` to define jet finder tasks to write to different tables based on inputs

Tests are ongoing in terms of functionality, but these changes should more or less complete an iteration in the development of the JE framework

**Review by iarsene — DISMISSED** [review]

File: Analysis/Core/include/AnalysisCore/JetUtilities.h [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    }
+    // Handle upper edge
+    if (jetsPhi[i] >= (2 * M_PI - (maxMatchingDistance + additionalMargin))) {
+      jetsPhiComparison.emplace_back(jetsPhi[i] - 2 * M_PI);
+      jetsEtaComparison.emplace_back(jetsEta[i]);
+      jetsMapToJetIndex.emplace_back(jetsMapToJetIndex[i]);
+    }
+  }
+
+  return std::move(std::make_tuple(jetsMapToJetIndex, jetsPhiComparison, jetsEtaComparison));
```
Comment (anerokhi) [inline Analysis/Core/include/AnalysisCore/JetUtilities.h:79]: Why not just `return { jetsMapToJetIndex, jetsPhiComparison, jetsEtaComparison };`?

**Comment by jgrosseo** [comment]
There seems to be conflicts now...

**Comment by iarsene** [comment]
Hi @raymondEhlers,
The macOS test is not related, but the fullCI test is genuine. Please fix it.

**Comment by raymondEhlers** [comment]
Hi @iarsene , thanks, I had overlooked that issue. It should be fixed now (although my o2 install is currently rebuilding because it broke due to changing dependencies, so I'm not 100% - I suppose we'll see)

**Comment by iarsene** [comment]
The macOS and dataflow tests are not related, merging

**Pull request merged**